### ISSUE TITLE
Exception during parsing: 'TypeArgumentListImpl' is not a subtype of type 'Expression'

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#46655]
We don't know the code.

```exception
[handleCodeActionRequest] type 'TypeArgumentListImpl' is not a subtype of type 'Expression' in type cast : #0      AstBuilder.endCascade (package:analyzer/src/fasta/ast_builder.dart:701)
#1      Parser.parseCascadeExpression (package:_fe_analyzer_shared/src/parser/parser_impl.dart:5183)
#2      Parser._parsePrecedenceExpressionLoop (package:_fe_analyzer_shared/src/parser/parser_impl.dart:4881)
#3      Parser.parsePrecedenceExpression (package:_fe_analyzer_shared/src/parser/parser_impl.dart:4857)
#4      Parser.parseExpression (package:_fe_analyzer_shared/src/parser/parser_impl.dart:4780)
#5      Parser.parseExpressionFunctionBody (package:_fe_analyzer_shared/src/parser/parser_impl.dart:4456)
#6      Parser.parseFunctionBody (package:_fe_analyzer_shared/src/parser/parser_impl.dart:4381)
#7      Parser.parseTopLevelMethod (package:_fe_analyzer_shared/src/parser/parser_impl.dart:2950)
#8      Parser.parseTopLevelMemberImpl (package:_fe_analyzer_shared/src/parser/parser_impl.dart:2729)
#9      Parser.parseTopLevelDeclarationImpl (package:_fe_analyzer_shared/src/parser/parser_impl.dart:502)
#10     Parser.parseUnit (package:_fe_analyzer_shared/src/parser/parser_impl.dart:357)
#11     Parser.parseCompilationUnit2 (package:analyzer/src/generated/parser.dart:195)
#12     Parser.parseCompilationUnit (package:analyzer/src/generated/parser.dart:191)
#13     FileState.parse (package:analyzer/src/dart/micro/library_graph.dart:205)
#14     FileState.refresh.<anonymous closure> (package:analyzer/src/dart/micro/library_graph.dart:257)
#15     OperationPerformanceImpl.run (package:analyzer/src/util/performance/operation_performance.dart:140)
#16     FileState.refresh (package:analyzer/src/dart/micro/library_graph.dart:254)
#17     FileSystemState.getFileForPath.<anonymous closure> (package:analyzer/src/dart/micro/library_graph.dart:661)
```

I believe @stereotype441 mentioned that he fixed something related to cascades.

**Comment by stereotype441** [comment]
Based on the line numbers in the stack trace the error came from a version of the parser before ee38656c0bba6982fecda510fb5c41f35517081d but no earlier than 9c5ac0150458cb23d44be625c834f2a13237c302.  (In particular, if the exception had come from ee38656c0bba6982fecda510fb5c41f35517081d or later, the `#1` line of the stacktrace would have been 5186, not 5183).

Checking out the code prior to ee38656c0bba6982fecda510fb5c41f35517081d, I'm able to reproduce the stacktrace by trying to parse the expression `a..foo<int>`, with assertions disabled.  This is precisely the bug that ee38656c0bba6982fecda510fb5c41f35517081d fixes (#46635).  The only reason the stack trace doesn't match #46635 is because the stack trace there is with assertions enabled, so it crashes a little earlier.

So I'm reasonably confident this the same as bug #46635, fixed by ee38656c0bba6982fecda510fb5c41f35517081d.

If we keep seeing exceptions like this, the thing to watch for is the line number on the `#1` line of the stacktrace.  If that number is always 5183, then the exception is coming from code prior to ee38656c0bba6982fecda510fb5c41f35517081d, so it confirms my theory.  If we ever see this stacktrace but with the `#1` line pointing to line 5186, then that will mean that there is some other problem and we'll need to investigate more deeply.

For now I'm going to mark this issue as a duplicate.

**Comment by stereotype441** [comment]
Duplicate of #46635
